File: backend/app/schemas/v1/watchlist_item.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


StrategyType = Literal["covered_calls", "put_options", "spread_options"]


class WatchlistItemCreate(BaseModel):
    strategy_type: StrategyType
    contract: str = Field(..., min_length=1, max_length=64)
    ticker: str = Field(..., min_length=1, max_length=32)
    exchange: int


class WatchlistItemOut(BaseModel):
    id: int
    user_id: int
    strategy_type: StrategyType
    contract: str
    ticker: str
    exchange: int
    created_at: datetime

    class Config:
        orm_mode = True

File: frontend/src/types/watchlistItem.ts
export type WatchlistStrategyType = "covered_calls" | "put_options" | "long_calls" | "long_puts";

export type WatchlistItem = {
  id: number;
  user_id: number;
  strategy_type: WatchlistStrategyType;
  contract: string;
  ticker: string;
  exchange: number;
  created_at: string;
};

export type CreateWatchlistItemPayload = {
  strategy_type: WatchlistStrategyType;
  contract: string;
  ticker: string;
  exchange: number;
};
